api-tests/platform/targets/tgt_dev_apis_tfm_nrf/nspe/pal_driver_intf.c

```diff
@@ -0,0 +1,159 @@
+/** @file
+ * Copyright (c) 2019-2020, Arm Limited or its affiliates. All rights reserved.
+ * SPDX-License-Identifier : Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+**/
+
+#include "pal_common.h"
+#include "pal_nvmem.h"
+
+
+typedef void (*pal_uart_logger_t)(const char *fmt, int32_t data);
+static pal_uart_logger_t logger;
+
+/* Log the format string to UART. */
+pal_uart_logger_t pal_uart_get_logger(void);
+
+/* Initialize the timer with the given number of ticks. */
+void pal_timer_init_ns(uint32_t ticks);
+
+/* Start the timer. */
+void pal_timer_start_ns(void);
+
+/* Stop and resest the timer. */
+void pal_timer_stop_ns(void);
+
+/* Get the address of a free, word-aligned, 1K memory area. */
+uint32_t pal_nvmem_get_addr(void);
+
+/**
+    @brief    - This function initializes the UART
+    @param    - uart base addr
+    @return   - SUCCESS/FAILURE
+**/
+int pal_uart_init_ns(uint32_t uart_base_addr)
+{
+    (void)uart_base_addr;
+    logger = pal_uart_get_logger();
+    return logger ? PAL_STATUS_SUCCESS : PAL_STATUS_ERROR;
+}
+
+/**
+    @brief    - This function parses the input string and writes bytes into UART TX FIFO
+    @param    - str      : Input String
+              - data     : Value for format specifier
+    @return   - SUCCESS/FAILURE
+**/
+
+int pal_print_ns(const char *str, int32_t data)
+{
+    logger(str, data);
```

Comment:
Why not use here `tfm_log_printf()` directly, as the psoc64 platform does? https://github.com/ARM-software/psa-arch-tests/blob/e52d4663332ca209ec3b9a2fdd1887dfaf0a791d/api-tests/platform/targets/tgt_dev_apis_tfm_psoc64/nspe/pal_driver_intf.c#L41-L43

Reply:
Good point, done.